### ISSUE TITLE
Concatenate error when trying to train a two input one output network

### ISSUE THREAD

**Issue by LiframCastle** [keras-team/keras#13096]
Hi,
I'm trying to train a 2 input one output model using the Keras API. The model seems to be right and all but at the end when I try to train it shows an error when it tries to concatenate the two outputs of the branches. It says that a two-dimensional array is expected and it gets a (x,y,z) input.
Being 'x' my batch size and 'y' and 'z' the size of the image. I have already tried using a batch size of 1 and the same error happens.
I have also tried using add instead of concatenating but the same error shows.

> Error when checking model target: expected dense_4 to have 2 dimensions, but got array with shape (10, 376, 1241)

That's the error I get

**Comment by its-mayank** [comment]
can you share your piece of code here? @LiframCastle

**Comment by LiframCastle** [comment]
Sure, I think my error is on the generator but I'm not sure why:

`image_shape = (376,1241) 
batch_size = 10

import os
from glob import glob
import re
import numpy as np
from imageio import imread
from skimage.transform import resize

def generator(data_folder, image_shape, batch_size):#, DAflag=False):
    image0_paths = glob(os.path.join(data_folder, 'colored_0', '*.png'))
    image1_paths = {re.sub(r'(scene)_', '_', os.path.basename(path)): path
        for path in glob(os.path.join(data_folder, 'colored_1', '*_*.png'))}
    #image1_paths.sort()
    label_paths = {re.sub(r'(scene)_', '_', os.path.basename(path)): path
        for path in glob(os.path.join(data_folder, 'disp_noc', '*_*.png'))}
    
    #background_color = np.array([0.0, 0.0, 0.0])  
    np.random.shuffle(image0_paths)
    while 1: 
        for batch_i in range(0, len(image0_paths)):#, batch_size):
            images = []
            images1 = []
            gt_images = []
            for image_file in image0_paths[batch_i:batch_i+batch_size]:
                gt_image_file = label_paths[os.path.basename(image_file)]
                image_file1 = image1_paths[os.path.basename(image_file)]

                x = resize(imread(image_file), image_shape, mode='reflect', 
                           anti_aliasing=True)
                x1= resize(imread(image_file1), image_shape, mode='reflect', 
                           anti_aliasing=True)
                y = resize(imread(gt_image_file), image_shape, mode='reflect', 
                           anti_aliasing=False)
               
                images.append(x)
                images1.append(x1)
                gt_images.append(y)
            X = np.array(images)
            X1 = np.array(images1)
            Y = np.array(gt_images)
            yield [X, X1], Y`

**Comment by gowtham-kp** [comment]
Similar to the issue #[9233](https://github.com/keras-team/keras/issues/9233). As stated there, Your targets should have shape (num_samples, num_classes). You're passing an array of shape (10, 376, 1241).

**Comment by LiframCastle** [comment]
But how can I fix that? I changed my generator but now I get the error :

output of generator should be a tuple `(x, y, sample_weight)` or `(x, y)`. Found: None
